### PR TITLE
Fix boolean error in warning; error out if only one of `--version` and `--codename` is supplied for an ansible-core release

### PR DESCRIPTION
**Pull request by felixfontein**
Improves the situation for #104 a bit.

### CODE REVIEW

File: src/antsibull_changelog/cli.py [hunk 1]
```diff
@@ -486,6 +486,12 @@ def _get_version_and_codename(paths: PathsConfig, config: ChangelogConfig,
     codename: str | None = args.codename
 
     if not config.is_collection and not config.is_other_project and not (version and codename):
+        if version or codename:
+            # Only one is supplied: error out instead of simply ignoring that one
+            raise ChangelogError(  # pylint: disable=raise-missing-from
```
Comment (gotmax23) [inline src/antsibull_changelog/cli.py:491]: Is this noqa comment actually necessary? I got no errors when I removed it and ran pylint locally.

Comment (felixfontein) [reply src/antsibull_changelog/cli.py:491]: Ah, probably not, I copy'n'pasted it from below.

Is there a way to make pylint report such things?

```suggestion
            raise ChangelogError(
```

**Review by briantist — APPROVED** [review]

File: src/antsibull_changelog/cli.py (hunk 1 above)
Comment (gotmax23) [reply src/antsibull_changelog/cli.py]: > Is there a way to make pylint report such things?

It seems there is: https://pylint.readthedocs.io/en/latest/user_guide/messages/information/useless-suppression.html

**Review by gotmax23 — APPROVED** [review]
Ack

**Comment by gotmax23** [comment]
Thanks @felixfontein and @briantist!

**Pull request merged**